### ISSUE TITLE
Timeout exception when sending message to a subscrition

### ISSUE THREAD

**Issue by oumaroudickoorckestra** [Azure/azure-service-bus-dotnet#291]
I am having the following error:
Exception: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond ErrorCode: TimedOut Message handler encountered an exception Microsoft.Azure.ServiceBus.ServiceBusCommunicationException: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed
because connected host has failed to respond ErrorCode: TimedOut ---> System.Net.Sockets.SocketException: A connection attempt failed because the connected party did notproperly respond after a period of time, or established connection failed because connected host has failed to respond

I am using the version for .Net Core. I am sure the connectionstring to the service bus is correct because the same information is working when using WindowsAzure.ServiceBus package on a .Net Framework project.

Basically, i have downloaded the sample programs and i have updated the following lines at the top in the BasicSendReceiveUsingTopicSubscriptionClient project

const string ServiceBusConnectionString = "Endpoint=sb://namespace.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=lXsQsYA/j1STOTGk58m8taVLs0VqIJPwa40DlzAqMAs=";
const string TopicName = "w506_events";
const string SubscriptionName = "Notifications";

Am i missing something?

Thanks

**Comment by SeanFeldman** [comment]
Timeouts are expected to happen. Generally, it's [recommendation](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-exceptions) is to retry your operation when handling timeout exceptions.

**Comment by oumaroudickoorckestra** [comment]
I do have a retry policy in place but every retry it is throwing the same timeout exception.

**Comment by SeanFeldman** [comment]
`RetryPolicy` is enabled by default. I was talking about custom retries.

> Retry might help in some cases; add retry logic to code.

**Comment by oumaroudickoorckestra** [comment]
Ok thanks. I will check that.

**Comment by oumaroudickoorckestra** [comment]
Actually the issue was coming from our internal firewall. We had to open the ports 5671 and 5672 in the firewall to make it work.
Thanks
